db/version_set.cc

```diff
@@ -1935,10 +1951,18 @@ void Version::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
       // do a final merge of nullptr and operands;
       std::string* str_value =
           iter->value != nullptr ? iter->value->GetSelf() : nullptr;
+      ValueType base_type = kTypeMerge;
       *status = MergeHelper::TimedFullMerge(
-          merge_operator_, user_key, nullptr, iter->merge_context.GetOperands(),
-          str_value, info_log_, db_statistics_, env_,
-          nullptr /* result_operand */, true);
+          merge_operator_, user_key, base_type, nullptr,
+          iter->merge_context.GetOperands(), str_value, info_log_,
+          db_statistics_, env_, nullptr /* result_operand */, true);
+      if (base_type == kTypeBlobIndex) {
+        ROCKS_LOG_ERROR(info_log_, "Encounter unexpected blob index.");
+        *status = Status::NotSupported(
+            "Encounter unexpected blob index. Please open DB with "
+            "rocksdb::blob_db::BlobDB instead.");
+        return;
+      }
```

Comment:
ok, I didn't realize Titan doesn't support MultiGet (needs to make `is_blob` a vector of boolean for each key being query).